src/backends/tensorflow/tensorflow_backend_tf.cc

```diff
@@ -405,22 +412,21 @@ TensorImpl::Init()
     cudaPointerAttributes attributes;
     cudaError_t err = cudaPointerGetAttributes(&attributes, nonstring_base_);
     gpu_tensor_ =
-        ((err == cudaSuccess) &&
-         (attributes.memoryType == cudaMemoryTypeDevice));
+        ((err == cudaSuccess) && (attributes.type == cudaMemoryTypeDevice));
   }
 }
 
 const std::string&
 TensorImpl::String(size_t idx) const
 {
-  auto flat = tftensor_.flat<std::string>();
+  auto flat = tftensor_.flat<tensorflow::tstring>();
```

Comment:
Is this necessary? 
https://github.com/tensorflow/community/pull/91#issuecomment-484696619 From this discussion here I feel we do not need to replace all strings with tensorflow::tstring. Am I missing something?

Reply:
Yes this is necessary. The template doesn't have specialization for std::string, at least in TF2